### PR TITLE
Misc cleanup for DOM attributes

### PR DESCRIPTION
**Pull request by eefriedman**
No functional change; just cleanups related to parsing attributes.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8404)

### CODE REVIEW

**Comment by nox** [comment]
A minor nit, otherwise LGTM.

-S-awaiting-review +S-needs-code-changes 


---

Reviewed 2 of 2 files at r1, 1 of 1 files at r2, 4 of 4 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

<sup>**[components/script/dom/htmlbodyelement.rs, line 100 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2axZcEu6Hn1Imvl1nU)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmlbodyelement.rs#L100)):</sup>
I'm pretty sure the `&` is superfluous.


---

<sup>**[components/script/dom/htmlbodyelement.rs, line 110 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2axbkmShwQxLarvF5g)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmlbodyelement.rs#L110)):</sup>
Ditto.


---

<sup>**[components/script/dom/htmlbodyelement.rs, line 120 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2axcZ_50YkPK9XOzqA)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmlbodyelement.rs#L120)):</sup>
Ditto.


---

<sup>**[components/script/dom/htmlfontelement.rs, line 94 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2axeH08vW0p8ZAYhOU)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmlfontelement.rs#L94)):</sup>
Ditto.


---

<sup>**[components/script/dom/htmlfontelement.rs, line 104 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2axfO6HZJ5XlbBuar4)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmlfontelement.rs#L104)):</sup>
Ditto.


---

<sup>**[components/script/dom/htmlfontelement.rs, line 114 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2axg-bIDIY9b8xKlJF)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmlfontelement.rs#L114)):</sup>
Ditto.


---

<sup>**[components/script/dom/htmliframeelement.rs, line 208 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2axjXvSiuYS5qfR7f7)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmliframeelement.rs#L208)):</sup>
Ditto.


---

<sup>**[components/script/dom/htmliframeelement.rs, line 210 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2ay-Br6kCEvIA9RESO)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmliframeelement.rs#L210)):</sup>
We should probably file a followup to make this a variant of `AttrValue` too.


---

<sup>**[components/script/dom/htmliframeelement.rs, line 218 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2axk_u0rD0jH68N8Es)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmliframeelement.rs#L218)):</sup>
Ditto.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8404)

**Comment by eefriedman** [comment]
Review status: 1 of 6 files reviewed at latest revision, 9 unresolved discussions.

---

<sup>**[components/script/dom/htmliframeelement.rs, line 210 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2ay-Br6kCEvIA9RESO)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmliframeelement.rs#L210)):</sup>
Sure; I'll file it.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8404)

**Comment by frewsxcv** [comment]
Fail seems legit

```
  ▶ TIMEOUT [expected OK] /html/obsolete/requirements-for-implementations/other-elements-attributes-and-apis/document-color-02.html
  │ 
  │ thread 'LayoutWorker worker 1/6' panicked at 'Color not found', /Users/servo/buildbot/slave/mac-rel-wpt/build/components/script/dom/attr.rs:118
  │ stack backtrace:
  │    1:        0x10b7f9b48 - sys::backtrace::tracing::imp::write::h662793599f713c74T5s
  │    2:        0x10b7fb63f - panicking::log_panic::_<closure>::closure.38923
  │    3:        0x10b7fb0e9 - panicking::log_panic::h87af6096e0b6fa7c4Uw
  │    4:        0x10b7e9dd6 - sys_common::unwind::begin_unwind_inner::h1e338bc64cf56d9dr8r
  │    5:        0x10a4c0924 - sys_common::unwind::begin_unwind::begin_unwind::h11769312394538122404
  │    6:        0x10a7fe5fc - dom::htmlbodyelement::_<impl>::get_background_color::h96cce9c616bb54e5ktw
  │    7:        0x10a4ae39a - css::matching::_<impl>::match_element::h823c88ff0903b64c9Hz
  │    8:        0x10a493160 - traversal::_<impl>::process::h3858f81d9981cb5bvYx
  │    9:        0x10a492821 - parallel::_<impl>::run_parallel::h13d7b06c0342da40aet
  │   10:        0x10a493d7e - parallel::recalc_style::h5edbc18b903df6bcAet
  │   11:        0x10a43113e - boxed::_<impl>::call_box::call_box::h11765958047692383731
  │   12:        0x10a40d5c2 - sys_common::unwind::try::try_fn::try_fn::h6634675915793128526
  │   13:        0x10b7f9018 - __rust_try
  │   14:        0x10b7f64be - sys_common::unwind::try::inner_try::hc13d8e198528cd0fZ4r
  │   15:        0x10a40d767 - boxed::_<impl>::call_box::call_box::h5255149473090508000
  │   16:        0x10b7faa0d - sys::thread::_<impl>::new::thread_start::h906ddce3a93d4852ksw
  │   17:     0x7fff8416a059 - _pthread_body
  └   18:     0x7fff84169fd6 - _pthread_start
```

File: components/script/dom/htmlbodyelement.rs [hunk 1]
```diff
@@ -141,8 +145,9 @@ impl VirtualMethods for HTMLBodyElement {
     }
 
     fn parse_plain_attribute(&self, name: &Atom, value: DOMString) -> AttrValue {
-        match name {
-            &atom!("text") => AttrValue::from_legacy_color(value),
+        match *name {
+            atom!("bgcolor") |
```
Comment (frewsxcv) [inline components/script/dom/htmlbodyelement.rs]: Should this be `bgColor`?

File: components/script/dom/htmlbodyelement.rs [hunk 2]
```diff
@@ -94,16 +87,27 @@ impl HTMLBodyElementMethods for HTMLBodyElement {
...
 
-impl HTMLBodyElement {
-    pub fn get_background_color(&self) -> Option<RGBA> {
-        self.background_color.get()
+impl HTMLBodyElementLayoutHelpers for LayoutJS<HTMLBodyElement> {
+    #[allow(unsafe_code)]
+    fn get_background_color(&self) -> Option<RGBA> {
+        unsafe {
+            (&*self.upcast::<Element>().unsafe_get())
+                .get_attr_for_layout(&ns!(""), &atom!("bgcolor"))
```
Comment (frewsxcv) [inline components/script/dom/htmlbodyelement.rs]: Should this be `bgColor`?

I actually don't know, just pointing out one of these might be capitalized

Comment (eefriedman) [inline components/script/dom/htmlbodyelement.rs]: HTML attributes are all lower-case; `bgcolor` is correct.

(I tracked down the reason for the test failure; it actually isn't a mistake in htmlbodyelement.rs.)

**Comment by eefriedman** [comment]
@nox Updated with a fix for document.rs ; please double-check that the change makes sense.

**Comment by nox** [comment]
-S-awaiting-review +S-needs-code-changes +S-needs-rebase 


---

Reviewed 1 of 2 files at r1, 1 of 1 files at r2, 4 of 4 files at r3, 2 of 2 files at r4, 3 of 3 files at r5, 1 of 1 files at r6, 2 of 2 files at r7.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

<sup>**[components/script/dom/htmlbodyelement.rs, line 100 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8404#-K2axZcEu6Hn1Imvl1nU)** ([raw file](https://github.com/servo/servo/blob/2926fe5dc3f53c74b48fafb0ec82b17cc14bb04a/components/script/dom/htmlbodyelement.rs#L100)):</sup>
You forgot that one.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8404)

**Pull request merged**